### PR TITLE
Convert gasnet-ex shim initialization over to the GEX API

### PR DESCRIPTION
**Pull request by ronawho**
Start migrating our gasnet-ex shim over to the gasnet-ex (GEX) API. To start, just migrate initialization.

The main change is that gasnet now has explicit teams and endpoints. For now we're just using a single endpoint/team (along with segment/client) to mimic the old behavior, but over time this could be used to implement multi-endpoint (parallel comm/injection) and multiple teams (for better collectives between subsets of locales)

Beyond that, the AM handler table takes additional information about how entires are used -- as a request/reply, with short/med/long AMs, how many args are used, and a debug name. `gasnet_attach` has also been split into separate attach and register-handlers steps, though we call them in one after the other so there's no real difference to us.

Testing:
 - [x] gex-udp-everything: full suite
 - [x] gex-ibv-large: `release/examples runtime/configMatters`

Part of #14443

### CODE REVIEW

**Comment by ronawho** [comment]
cc @bonachea @PHHargrove

File: runtime/src/comm/gasnet/comm-gasnet-ex.c [hunk 1]
```diff
@@ -686,6 +690,7 @@ int chpl_comm_try_nb_some(chpl_comm_nb_handle_t* h, size_t nhandles)
   return gasnet_try_syncnb_some((gasnet_handle_t*) h, nhandles) == GASNET_OK;
 }
 
+// TODO GEX could be scalable query to gasnet itself
 int chpl_comm_addr_gettable(c_nodeid_t node, void* start, size_t len)
 {
 #ifdef GASNET_SEGMENT_EVERYTHING
```
Comment (bonachea) [inline runtime/src/comm/gasnet/comm-gasnet-ex.c:696]: Orthogonal to this PR, but an observation in reading the code below: 

I think this function can make/assert the reasonable assumption that `reqstart <= reqend`, IOW the `len` value is valid and not a ridiculously enormous value that indicates a request that wraps around the top of the address space (if that were violated, then the caller has already gone very wrong before reaching here).

With that assumption, then only two of the comparison operations below are actually required (the first and last). The other two are implied by this property and `segstart < segend` (an invariant guaranteed by GASNet). Further, I believe the C optimizer lacks the invariant knowledge required to automate this optimization, which could save a few instructions per call.

I believe this observation applies equally to both this version of the code using the GASNet-1-style `seginfo_table[]`, and a future GEX version of the code that calls `gex_EP_QueryBoundSegmentNB()` to retrieve the bounds from GASNet.

**Review by bonachea — APPROVED** [review]
LGTM!

Added one orthogonal observation for a simple future improvement.

File: runtime/src/comm/gasnet/comm-gasnet-ex.c (hunk 1 above)
Comment (ronawho) [reply runtime/src/comm/gasnet/comm-gasnet-ex.c:696]: Agreed, I've added a TODO for this to https://github.com/chapel-lang/chapel/issues/14443#issuecomment-1795227749

**Review by jhh67 — APPROVED** [review]

**Pull request merged**